AND-SOPT-iOS/AND-SOPT-iOS/Assignment/Week02-03/FinanceView/TopFree.swift

```diff
@@ -0,0 +1,127 @@
+        return button
+    }()
+    
+    private lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+        collectionView.isPagingEnabled = true
+        collectionView.showsHorizontalScrollIndicator = false
+        return collectionView
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setUI()
+        setLayout()
+        setCollectionView()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setCollectionView() {
+        let flowLayout = UICollectionViewFlowLayout()
+        flowLayout.scrollDirection = .horizontal
+        flowLayout.itemSize = CGSize(width: UIScreen.main.bounds.width, height: 80)
+        flowLayout.minimumLineSpacing = 0
+        
+        collectionView.do {
+            $0.setCollectionViewLayout(flowLayout, animated: true)
+            $0.register(TopPaidFreeAppCell.self, forCellWithReuseIdentifier: TopPaidFreeAppCell.identifier)
+            $0.delegate = self
+            $0.dataSource = self
+        }
+    }
+    
+    private func setUI() {
+        addSubviews(topFreeButton, collectionView)
+    }
+    
+    private func setLayout() {
+        topFreeButton.snp.makeConstraints {
+            $0.top.equalToSuperview()
+            $0.leading.equalToSuperview().offset(20)
+        }
+        
+        collectionView.snp.makeConstraints {
+            $0.top.equalTo(topFreeButton.snp.bottom).offset(12)
+            $0.horizontalEdges.equalToSuperview()
+            $0.height.equalTo(265)
+        }
+    }
+    
+    @objc private func topFreeButtonTapped() {
+        delegate?.didTapTopFreeButton()
+    }
+}
+
+extension TopFree: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if indexPath.row == 8 {
+            delegate?.didTapTossCell()
+        }
+    }
+}
+
+extension TopFree: UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 9
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let item = collectionView.dequeueReusableCell(withReuseIdentifier: TopPaidFreeAppCell.identifier, for: indexPath) as? TopPaidFreeAppCell else {
+            return UICollectionViewCell()
+        }
+        
+        item.bind(appList[indexPath.item])
+        
+        let dividerLine = UIView()
+        dividerLine.backgroundColor = .systemGray
+        
+        if indexPath.row % 3 != 2 {
+            item.addSubview(dividerLine)
+            dividerLine.snp.makeConstraints {
+                $0.leading.equalToSuperview().offset(80)
+                $0.trailing.equalToSuperview().offset(-20)
```

Comment:
```suggestion
                $0.trailing.equalToSuperview().inset(20)
```
offset(-20)보단 inset(20)이라고 쓰시는 게 좀 더 직관적일 것 같습니다.

Reply:
참고하겠습니다!